hudi-common/src/main/java/org/apache/hudi/common/table/read/IncrementalQueryAnalyzer.java

```diff
@@ -471,5 +500,20 @@ public HoodieTimeline getActiveTimeline() {
     public @Nullable HoodieTimeline getArchivedTimeline() {
       return archivedTimeline;
     }
+
+    public Option<String> getPredicateFilter() {
+      return predicateFilter;
+    }
+
+    public QueryContext withUpdatedEndInstant(String endInstant) {
+      return new QueryContext(
+          startInstant.get(),
+          endInstant,
+          instants,
+          archivedInstants,
+          activeInstants,
+          activeTimeline,archivedTimeline,
+          null);
+    }
```

Comment:
Should we avoid this API to disallow changing the result?

Reply:
Makes sense. I'll remove this method